ocaml/idl/datamodel.ml

```diff
@@ -9209,6 +9211,61 @@ let alert =
     ()
 *)
 
+
+(** network sriov **)
+module Network_sriov = struct
+  let lifecycle = [Published, rel_kolkata, ""]
+
+  let configuration_mode = Enum ("configuration_mode",
+    [
+      "sysfs", "Configure network sriov by sysfs, do not need reboot";
+      "modprobe", "Configure network sriov by modbrope, need reboot";
+      "unknown", "Unknown mode";
+    ])
+
+  let create = call
+      ~name:"create"
+      ~doc:"Create a network-sriov based on the specific PIF, it will automatically create a logical PIF to connect the specific network."
+      ~params:[Ref _pif, "pif", "PIF on which to enable SRIOV";
+               Ref _network, "network", "Network to link SRIOV"]
+      ~result:(Ref _network_sriov, "The reference of the created SRIOV object")
+      ~lifecycle
+      ~allowed_roles:_R_POOL_OP
+      ()
+
+  let destroy = call
+      ~name:"destroy"
+      ~doc:"Destroy a network-sriov, will destroy the logical PIF accordingly."
+      ~params:[Ref _network_sriov, "self", "SRIOV to destroy"]
+      ~lifecycle
+      ~allowed_roles:_R_POOL_OP
+      ()
+
+  let obj =
+    create_obj
+      ~name:_network_sriov
+      ~descr:"network-sriov which connects logical pif and physical pif"
+      ~doccomments:[]
+      ~gen_constructor_destructor:false
+      ~gen_events:true
+      ~in_db:true
+      ~lifecycle
+      ~messages:[create; destroy;]
+      ~messages_default_allowed_roles:_R_POOL_OP
+      ~persist:PersistEverything
+      ~in_oss_since:None
+      ~contents:
+        ([
+          uid _network_sriov;
+          field ~qualifier:StaticRO ~ty:(Ref _pif) ~lifecycle "physical_PIF" "physical PIF which the network_sriov created" ~default_value:(Some (VRef ""));
+          field ~qualifier:StaticRO ~ty:(Ref _pif) ~lifecycle "logical_PIF" "logical PIF after sriov enabled for physical pif" ~default_value:(Some (VRef ""));
```

Comment:
This should also be a bit more descriptive.

Reply:
ok